src/ucp/core/ucp_worker.c

```diff
@@ -171,7 +171,8 @@ static void ucp_worker_wakeup_context_cleanup(ucp_worker_wakeup_t *wakeup)
 }
 
 static ucs_status_t ucp_worker_add_iface(ucp_worker_h worker,
-                                         ucp_rsc_index_t tl_id)
+                                         ucp_rsc_index_t tl_id,
+                                         ucs_cpu_set_t const * cpu_set_param)
```

Comment:
I guess once we introduce memory affinity the interface will change again ? I suggest generalize the interface such that you path the arguments down the stack.